android/app/src/main/java/com/example/secretgenerator/MainActivity.kt

```diff
@@ -30,15 +33,33 @@ class MainActivity : AppCompatActivity() {
         }
 
         moveButton.setOnClickListener {
-            val intent = Intent(this@MainActivity, SecretActivity::class.java)
-            intent.putExtra("publicKey", publicKeyTextView.text.toString())
-            startActivityForResult(intent, REQUEST_SECRET_KEY)
+            if (publicKeyTextView.text.isNotEmpty()) {
+                val intent = Intent(this@MainActivity, SecretActivity::class.java)
+                intent.putExtra(KEY, publicKeyTextView.text.toString())
+                startActivityForResult(intent, REQUEST_SECRET_KEY)
+                publicKeyTextView.text = ""
+            } else
+                Toast.makeText(this@MainActivity, "YOU SHOULD GENERATE KEY", Toast.LENGTH_SHORT)
+                    .show()
         }
     }
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
```

Comment:
Deprecated function. But Kamil wants me to use onActivityResult. Maybe I should provide better solution? (Contracts)